CRM/Financial/BAO/Payment.php

```diff
@@ -41,8 +42,11 @@ class CRM_Financial_BAO_Payment {
    * @throws \CRM_Core_Exception
    */
   public static function create(array $params): CRM_Financial_DAO_FinancialTrxn {
-    $contribution = civicrm_api3('Contribution', 'getsingle', ['id' => $params['contribution_id']]);
-    $contributionStatus = CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'contribution_status_id', $contribution['contribution_status_id']);
+    $contribution = Contribution::get(FALSE)
+      ->addWhere('id', '=', $params['contribution_id'])
+      ->addSelect('*', 'contribution_status_id:name', 'balance_amount', 'paid_amount')
```

Comment:
'balance_amount', 'paid_amount'... how are these being used? Is it something v3 had been returning previously?

Reply:
nah - there are some separate functions that can use these instead of calling some other functions if they are present - but that change requires another array to be wrangled